syntax = "proto3";
package service;
option java_package = "org.qmstr.grpc.service";

import weak "datamodel.proto";

message AnalyzerConfigRequest {
    int32 analyzerID = 1;
}

message AnalyzerConfigResponse {
    string analyzerConfig = 1;
    int64 token = 2;
}

message InfoNodesMessage {
    int64 token = 1;
    string uid = 2;
    repeated InfoNode infonodes = 3;
}

message DiagnosticNodeMessage {
    int64 token = 1;
    string uid = 2;
    DiagnosticNode diagnosticnode =3;
}
message SendResponse {
    bool success = 1;
}

message DummyRequest {
}

service AnalysisService {
    rpc GetAnalyzerConfig(AnalyzerConfigRequest) returns (AnalyzerConfigResponse) {}
    rpc SendInfoNodes(stream InfoNodesMessage) returns (SendResponse) {}
    rpc SendDiagnosticNode(stream DiagnosticNodeMessage) returns (SendResponse) {}
    rpc GetSourceFileNodes(DummyRequest) returns (stream FileNode) {}
}